Document module-level statistics in SolarEdge (#40922)
home-assistant/home-assistant.github.io: source/_integrations/solaredge.markdown

ha_codeowners:
  - '@frenck'
  - '@bdraco'
  - '@tronikos'
ha_integration_type: device
---

The SolarEdge integration allows you to get details from your SolarEdge solar power setup and integrate these in your Home Assistant installation.
It provides two main features:

> 1.  **Site Sensors**: Provides real-time energy data for your entire site (e.g., current power, energy today, lifetime energy). This requires a site ID and an API Key.

Reviewer: ```suggestion
1.  Site sensors: Provide real-time energy data for your entire site. For example, current power, energy today, and lifetime energy. This requires a site ID and an API Key.
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization). Please apply throughout the docs.

> 2.  **Module-level Statistics**: Retrieves energy production data per inverter, string, and module, and inserts it into Home Assistant statistics. This is useful for identifying underperforming modules. This requires a site ID, username, and password.

Reviewer: ```suggestion
2.  Module-level statistics: Retrieves energy production data per inverter, string, and module, and inserts it into Home Assistant statistics. This is useful for identifying underperforming modules. This requires a site ID, username, and password.
```

You can configure either feature individually or both at the same time for the same site.

{% include integrations/config_flow.md %}

> ## Site Sensors (API Key)

Reviewer: ```suggestion
## Site sensors (API key)
```

    JOIN statistics_meta sm ON s.metadata_id = sm.id
    CROSS JOIN RelevantTimeRange rt
    WHERE
        sm.source = 'solaredge'
        -- TODO: Adjust this to match your setup
        AND sm.name LIKE '% 1.1.%'
        AND s.start_ts >= rt.period_start_ts
        AND s.start_ts < rt.period_end_ts
        AND s.state IS NOT NULL
    GROUP BY
        sm.name
),
AverageProduction AS (
    -- Calculate the average production across all modules that produced something
    SELECT
        AVG(mp.total_production) AS average_total_production
    FROM ModuleProduction mp
    WHERE total_production > 0
)
SELECT
    IFNULL(
        GROUP_CONCAT(
            mp.name || ' (' || printf("%.1f", (mp.total_production * 100.0 / ap.average_total_production)) || '%)',
            ', '
        ),
        ''
    ) AS problematic_modules
FROM
    ModuleProduction mp, AverageProduction ap -- Implicit cross join, AP will have 1 row
WHERE
    -- TODO: Adjust the 95% threshold if needed
    mp.total_production < (0.95 * ap.average_total_production)
    AND ap.average_total_production IS NOT NULL
    AND ap.average_total_production > 0 -- Avoid division by zero if no production at all
) AS result
```

> This will result in a sensor with state ,e.g., `SolarEdge 1.1.13 (94.7%), SolarEdge 1.1.14 (94.2%)`

Reviewer: ```suggestion
This will result in a sensor with a state. For example, `SolarEdge 1.1.13 (94.7%), SolarEdge 1.1.14 (94.2%)`
```
